server.go

```diff
@@ -222,6 +245,11 @@ func (server *Server) handle(s *streamWrap) error {
 
 	replyv = reflect.New(mtype.ReplyType.Elem())
 
+	if server.Authorize == nil || !server.Authorize(s.stream.Conn().RemotePeer(), svcID.Name, svcID.Method) {
+		resp := &Response{svcID, "client does not have permissions to this method, service name: " + svcID.Name + ", method name: " + svcID.Method, serverErr}
```

Comment:
This is not wrong per se, but try to keep lines shorter and preferentially use `fmt.Printf()` to concatenating strings.

Reply:
`errorMsg := fmt.Printf(...)` does not hurt.

Reply:
You mean fmt.SPrintf(...) ?

Reply:
yes, sorry